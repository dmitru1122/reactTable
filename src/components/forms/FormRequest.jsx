import React, { useState, useEffect } from 'react';
import { Form, Field } from 'react-final-form';
import { TextInput, Select, SelectItem, RadioButtonGroup, RadioButton, Button } from 'carbon-components-react';
import { Container, Row, Col } from 'reactstrap';
import './FormRequest.scss';
import PropTypes from 'prop-types';

const selectItemList = [
  { text: 'Job proposition', value: 'Job proposition' },
  { text: 'Review', value: 'Review' },
  { text: 'Another', value: 'Another' },
];

const FormRequest = (props) => {
  const { title, initialData, action } = props;
  const [formData, setFormData] = useState({ initialData });
  const [isValid, setIsValid] = useState(false);
  const onSubmit = (values, form) => {
    action(values, form);
  };

  useEffect(() => {
    setFormData(initialData);
  }, [initialData]);

  return (
    <Container className='form'>
      {title && <h2 className='form__title'>{title}</h2>}
      <Form
        onSubmit={onSubmit}
        initialValues={formData}
        validate={(values) => {
          if (values.lastName && values.firstName && values.gender && values.purpose) {
            setIsValid(true);
          } else {
            setIsValid(false);
          }
        }}
        render={({ handleSubmit, form, submitting, pristine, values }) => (
          <form onSubmit={handleSubmit} data-testid='form'>
            <Row>
              <Col className='col-md-12'>
                <Field name='purpose'>
                  {({ input, meta }) => (
                    <div className='form__field'>
                      <Select
                        {...input}
                        required
                        id='select-1'
                        data-testid='purpose-select'
                        invalidText='This is an invalid error message.'
                        invalid={meta.error && meta.touched}
                        labelText='Select purpose of the request'>
                        <SelectItem text={values.purpose || 'Purpose'} value={values.purpose || ''} disabled />
                        {selectItemList.map((item) => (
                          <SelectItem text={item.text} key={item.text} value={item.value} />
                        ))}
                      </Select>
                    </div>
                  )}
                </Field>
              </Col>
            </Row>
            <Row>
              <Col className='form__field col-sm-6 col-12'>
                <Field name='firstName'>
                  {({ input, meta }) => (
                    <TextInput
                      {...input}
                      required
                      className='bx--text-input--lg'
                      labelText=''
                      id='firstName'
                      invalidText='Input First Name.'
                      invalid={meta.error && meta.touched}
                      placeholder='First Name'
                    />
                  )}
                </Field>
              </Col>
              <Col className='form__field col-sm-6 col-12'>
                <Field name='lastName'>
                  {({ input, meta }) => (
                    <TextInput
                      {...input}
                      required
                      labelText=''
                      className='bx--text-input--lg'
                      invalid={meta.error && meta.touched}
                      id='test2'
                      invalidText='Input Last Name.'
                      placeholder='Last Name'
                    />
                  )}
                </Field>
              </Col>
            </Row>
            <Row>
              <Col>
                <Field name='gender'>
                  {({ input, meta }) => (
                    <div className='form__field'>
                      <h4 htmlFor='male'>Gender</h4>
                      <RadioButtonGroup
                        {...input}
                        invalid={meta.error && meta.touched}
                        invalidText='This is an invalid error message.'
                        defaultSelected={values.gender}
                        legend='Group Legend'
                        labelText='Last Name'
                        valueSelected={values.gender}
                        name='radio-button-group'>
                        <RadioButton id='male' labelText='male' value='male' />
                        <RadioButton id='female' labelText='female' value='female' />
                        <RadioButton required id='another' labelText='another' value='another' />
                      </RadioButtonGroup>
                    </div>
                  )}
                </Field>
              </Col>
            </Row>
            <Row className='justify-content-center'>
              <Col className='d-flex form__field'>
                <Button
                  kind={!isValid ? 'danger--tertiary' : 'tertiary'}
                  type='submit'
                  className={`form__submit-btn ${!isValid && 'button--disabled bx--btn--danger--tertiary'}`}>
                  {!isValid ? 'Disabled' : 'Submit'}
                </Button>
              </Col>
              <Col className='d-flex form__field'>
                <Button
                  kind={submitting || pristine ? 'danger--tertiary' : 'tertiary'}
                  disabled={submitting || pristine}
                  className={`form__reset-btn ${
                    (submitting || pristine) && 'button--disabled bx--btn--danger--tertiary'
                  }`}
                  onClick={form.reset}
                  type='button'>
                  Reset
                </Button>
              </Col>
            </Row>
          </form>
        )}
      />
    </Container>
  );
};

FormRequest.propTypes = {
  title: PropTypes.string,
  initialData: PropTypes.shape({
    id: PropTypes.string,
    purpose: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    gender: PropTypes.string,
  }),
  action: PropTypes.func,
};
FormRequest.defaultProps = {
  title: null,
  initialData: { id: '', purpose: '', firstName: '', lastName: '', gender: '' },
  action: null,
};

export default FormRequest;
