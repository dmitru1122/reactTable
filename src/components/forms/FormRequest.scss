@import '../../theme/variables.scss';

.button {
  &--disabled {
    color: rgba(16, 16, 16, 0.3);
    background-color: rgba(239, 239, 239, 0.3);
    border-color: rgba(195, 195, 195, 0.3);
    cursor: not-allowed !important;
  }
}

.form {
  &__submit-btn {
    margin-left: auto;
  }
  &__reset-btn {
    margin-right: auto;
  }

  @media screen and (max-width: $breakpoint-xs) {
    &__reset-btn,
    &__submit-btn {
      margin-left: auto;
      margin-right: auto;
    }
  }

  &__field {
    margin-bottom: 10px;
  }
}
