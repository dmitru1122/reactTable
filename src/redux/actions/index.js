import actionTypes from './actionTypes';

export function failure(error) {
  return {
    type: actionTypes.FAILURE,
    error,
  };
}

export function loadData() {
  return { type: actionTypes.LOAD_DATA };
}

export function loadDataSuccess(data, id) {
  return {
    type: actionTypes.LOAD_DATA_SUCCESS,
    data,
    id,
  };
}

export function addOneRequest(data) {
  return {
    type: actionTypes.ADD_ONE_REQUEST,
    data,
  };
}

// export function addOneRequestSuccess(data) {
//   return {
//     type: actionTypes.ADD_ONE_REQUEST_SUCCESS,
//     data,
//   };
// }
// export function addOneRequestFail() {
//   return {
//     type: actionTypes.ADD_ONE_REQUEST_FAIL,
//   };
// }

export function loadOneRequest(id) {
  return { type: actionTypes.LOAD_ONE_REQUEST, id };
}

export function loadOneRequestSuccess(data, id) {
  return {
    type: actionTypes.LOAD_ONE_REQUEST_SUCCESS,
    data,
    id,
  };
}

export function deleteOneRequest(id) {
  return { type: actionTypes.DELETE_ONE_REQUEST, id };
}
export function deleteOneRequestReset() {
  return { type: actionTypes.DELETE_ONE_REQUEST_RESET };
}

export function deleteOneRequestSuccess(id) {
  return { type: actionTypes.DELETE_ONE_REQUEST_SUCCESS, id };
}
export function deleteOneRequestFail() {
  return { type: actionTypes.DELETE_ONE_REQUEST_FAIL };
}

export function editOneRequest(data, id) {
  return { type: actionTypes.EDIT_ONE_REQUEST, data, id };
}
// export function editOneRequestSuccess(data, id) {
//   return { type: actionTypes.EDIT_ONE_REQUEST_SUCCESS, data, id };
// }
// export function editOneRequestFail() {
//   return { type: actionTypes.EDIT_ONE_REQUEST_FAIL };
// }
