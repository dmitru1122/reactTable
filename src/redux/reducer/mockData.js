export const initialState = {
  error: null,
  lastUpdate: 0,
  requestData: { headerData: null, rowData: null },
  deleteRequestStatus: 'waiting',
  fullRequestInfo: {},
  placeholderData: null,
};

export const rowData = [{ firstName: 'Vit', gender: 'male', id: '2', lastName: 'vab', purpose: 'Job proposition' }];
export const rowDataAfterFilter = [{ firstName: 'Vit', id: '2', purpose: 'Job proposition' }];
export const headerData = [
  { key: 'purpose', header: 'Purpose' },
  { key: 'firstName', header: 'First Name' },
];
export const fullHeaderData = [
  { key: 'purpose', header: 'Purpose' },
  { key: 'firstName', header: 'First Name' },
  { key: 'lastName', header: 'Last Name' },
  { key: 'gender', header: 'Purpose' },
];

export const failureMock = {
  data: {
    headerData,
    rowData: rowDataAfterFilter,
  },
  id: '2',
};

export const loadDataMock = {
  data: {
    headerData,
    rowData,
  },
  id: '2',
};
export const loadOneRequestMock = {
  data: {
    headerData: fullHeaderData,
    rowData,
  },
  id: '2',
};
export const loadSecondOneRequestMock = {
  data: {
    headerData: fullHeaderData,
    rowData: [{ firstName: 'Dima', gender: 'male', id: '3', lastName: 'Pashkevich', purpose: 'Job proposition' }],
  },
  id: '3',
};

export const addOneRequestDataMock = rowData;
//  {
//   firstName: 'Inna',
//   gender: 'male',
//   id: '',
//   lastName: 'test',
//   purpose: 'Job proposition',
// };

export const editOneRequestMock = {
  data: rowData,
  id: '2',
};
