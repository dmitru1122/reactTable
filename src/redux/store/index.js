import { applyMiddleware, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducer/allReducer';
import rootSaga from '../saga/saga';

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    return composeWithDevTools(applyMiddleware(...middleware));
  }
  return applyMiddleware(...middleware);
};

export const makeStore = () => {
  const sagaMiddleware = createSagaMiddleware();

  const store = createStore(rootReducer, bindMiddleware([sagaMiddleware]));

  store.sagaTask = sagaMiddleware.run(rootSaga);

  return store;
};

export default makeStore();
