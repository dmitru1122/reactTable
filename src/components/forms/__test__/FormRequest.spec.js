import React from 'react';
import { mount } from 'enzyme';
import FormRequest from '../FormRequest';
import { rowData } from '../../../redux/mockData';

const setUp = (props) => mount(<FormRequest {...props} />);

const props = {
  title: 'Test title',
  initialData: { ...rowData[0] },
  action: () => {},
};

describe('FormRequest', () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });

  it('shold render button', () => {
    expect(component).toBeTruthy();
  });
  it('should create shanapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should get props', () => {
    expect(component.prop('title')).toBe(props.title);
    expect(component.prop('initialData')).toBe(props.initialData);
    expect(component.prop('action')).toBe(props.action);
  });

  it('should render ReturnToListButton', () => {
    expect(component.find('[data-testid="form"]')).toBeTruthy();
  });

  describe('Field purpose', () => {
    it('should render without crashing', () => {
      expect(component.find('Field [name="purpose"]')).toBeTruthy();
    });

    it('should has props value', () => {
      expect(component.find('[name="purpose"]').find('select').prop('value')).toBe(props.initialData.purpose);
    });

    it('should handle input value', () => {
      const expectedValue = 'test purpose';
      const element = component.find('[name="purpose"]').find('select');
      element.simulate('change', { target: { name: 'purpose', value: expectedValue } });
      expect(component.find('[name="purpose"]').find('select').prop('value')).toBe(expectedValue);
    });
  });

  describe('Field firstName', () => {
    it('should render without crashing', () => {
      expect(component.find('Field [name="firstName"]')).toBeTruthy();
    });

    it('should has props value', () => {
      expect(component.find('[name="firstName"]').find('input').prop('value')).toBe(props.initialData.firstName);
    });

    it('should handle input value', () => {
      const expectedValue = 'test first name';
      const element = component.find('[name="firstName"]').find('input');
      element.simulate('change', { target: { name: 'firstName', value: expectedValue } });
      expect(component.find('[name="firstName"]').find('input').prop('value')).toBe(expectedValue);
    });

    describe('Field lastName', () => {
      it('should render without crashing', () => {
        expect(component.find('Field [name="lastName"]')).toBeTruthy();
      });

      it('should has props value', () => {
        expect(component.find('[name="lastName"]').find('input').prop('value')).toBe(props.initialData.lastName);
      });

      it('should handle input value', () => {
        const expectedValue = 'test last name';
        const element = component.find('[name="lastName"]').find('input');
        element.simulate('change', { target: { name: 'lastName', value: expectedValue } });
        expect(component.find('[name="lastName"]').find('input').prop('value')).toBe(expectedValue);
      });
    });

    describe('Field gender', () => {
      it('should render without crashing', () => {
        expect(component.find('Field [name="gender"]')).toBeTruthy();
      });

      it('should has props value', () => {
        expect(component.find('[name="gender"]').find('RadioButtonGroup').prop('value')).toBe(props.initialData.gender);
      });

      it('should handle input value', () => {
        const expectedActiveId = 'female';
        const element = component
          .find('[name="gender"]')
          .find('RadioButtonGroup')
          .find(`[id="${expectedActiveId}"]`)
          .find('input');
        element.simulate('change', { target: { checked: true } });
        expect(component.find('[name="gender"]').find('RadioButtonGroup').prop('value')).toBe(expectedActiveId);
      });
    });
  });
});
