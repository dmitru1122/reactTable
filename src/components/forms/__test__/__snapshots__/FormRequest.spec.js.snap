// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRequest should create shanapshot 1`] = `
<FormRequest
  action={[Function]}
  initialData={
    Object {
      "firstName": "Vit",
      "gender": "male",
      "id": "2",
      "lastName": "vab",
      "purpose": "Job proposition",
    }
  }
  title="Test title"
>
  <Container
    className="form"
    tag="div"
  >
    <div
      className="form container"
    >
      <h2
        className="form__title"
      >
        Test title
      </h2>
      <ReactFinalForm
        initialValues={
          Object {
            "firstName": "Vit",
            "gender": "male",
            "id": "2",
            "lastName": "vab",
            "purpose": "Job proposition",
          }
        }
        onSubmit={[Function]}
        render={[Function]}
        validate={[Function]}
      >
        <form
          data-testid="form"
          onSubmit={[Function]}
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <Col
                className="col-md-12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-12 col"
                >
                  <ForwardRef(Field)
                    name="purpose"
                  >
                    <div
                      className="form__field"
                    >
                      <Select
                        data-testid="purpose-select"
                        disabled={false}
                        helperText=""
                        id="select-1"
                        inline={false}
                        invalid={false}
                        invalidText="This is an invalid error message."
                        labelText="Select purpose of the request"
                        light={false}
                        name="purpose"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={true}
                        value="Job proposition"
                      >
                        <div
                          className="bx--form-item"
                        >
                          <div
                            className="bx--select"
                          >
                            <label
                              className="bx--label"
                              htmlFor="select-1"
                            >
                              Select purpose of the request
                            </label>
                            <div
                              className="bx--select-input__wrapper"
                              data-invalid={null}
                            >
                              <select
                                className="bx--select-input"
                                data-testid="purpose-select"
                                id="select-1"
                                name="purpose"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={true}
                                value="Job proposition"
                              >
                                <SelectItem
                                  disabled={true}
                                  hidden={false}
                                  text="Job proposition"
                                  value="Job proposition"
                                >
                                  <option
                                    className="bx--select-option"
                                    disabled={true}
                                    hidden={false}
                                    value="Job proposition"
                                  >
                                    Job proposition
                                  </option>
                                </SelectItem>
                                <SelectItem
                                  disabled={false}
                                  hidden={false}
                                  key="Job proposition"
                                  text="Job proposition"
                                  value="Job proposition"
                                >
                                  <option
                                    className="bx--select-option"
                                    disabled={false}
                                    hidden={false}
                                    value="Job proposition"
                                  >
                                    Job proposition
                                  </option>
                                </SelectItem>
                                <SelectItem
                                  disabled={false}
                                  hidden={false}
                                  key="Review"
                                  text="Review"
                                  value="Review"
                                >
                                  <option
                                    className="bx--select-option"
                                    disabled={false}
                                    hidden={false}
                                    value="Review"
                                  >
                                    Review
                                  </option>
                                </SelectItem>
                                <SelectItem
                                  disabled={false}
                                  hidden={false}
                                  key="Another"
                                  text="Another"
                                  value="Another"
                                >
                                  <option
                                    className="bx--select-option"
                                    disabled={false}
                                    hidden={false}
                                    value="Another"
                                  >
                                    Another
                                  </option>
                                </SelectItem>
                              </select>
                              <ForwardRef(ChevronDown16)
                                className="bx--select__arrow"
                              >
                                <Icon
                                  className="bx--select__arrow"
                                  fill="currentColor"
                                  height={16}
                                  preserveAspectRatio="xMidYMid meet"
                                  viewBox="0 0 16 16"
                                  width={16}
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="bx--select__arrow"
                                    fill="currentColor"
                                    focusable="false"
                                    height={16}
                                    preserveAspectRatio="xMidYMid meet"
                                    viewBox="0 0 16 16"
                                    width={16}
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                                    />
                                  </svg>
                                </Icon>
                              </ForwardRef(ChevronDown16)>
                            </div>
                          </div>
                        </div>
                      </Select>
                    </div>
                  </ForwardRef(Field)>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <Col
                className="form__field col-sm-6 col-12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="form__field col-sm-6 col-12 col"
                >
                  <ForwardRef(Field)
                    name="firstName"
                  >
                    <ForwardRef(TextInput)
                      className="bx--text-input--lg"
                      disabled={false}
                      helperText=""
                      id="firstName"
                      inline={false}
                      invalid={false}
                      invalidText="Input First Name."
                      labelText=""
                      light={false}
                      name="firstName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      placeholder="First Name"
                      required={true}
                      type="text"
                      value="Vit"
                      warn={false}
                      warnText=""
                    >
                      <div
                        className="bx--form-item bx--text-input-wrapper"
                      >
                        <div
                          className="bx--text-input__field-outer-wrapper"
                        >
                          <div
                            className="bx--text-input__field-wrapper"
                            data-invalid={null}
                          >
                            <input
                              className="bx--text-input bx--text-input--lg"
                              disabled={false}
                              id="firstName"
                              name="firstName"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              placeholder="First Name"
                              required={true}
                              title="First Name"
                              type="text"
                              value="Vit"
                            />
                          </div>
                        </div>
                      </div>
                    </ForwardRef(TextInput)>
                  </ForwardRef(Field)>
                </div>
              </Col>
              <Col
                className="form__field col-sm-6 col-12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="form__field col-sm-6 col-12 col"
                >
                  <ForwardRef(Field)
                    name="lastName"
                  >
                    <ForwardRef(TextInput)
                      className="bx--text-input--lg"
                      disabled={false}
                      helperText=""
                      id="test2"
                      inline={false}
                      invalid={false}
                      invalidText="Input Last Name."
                      labelText=""
                      light={false}
                      name="lastName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      placeholder="Last Name"
                      required={true}
                      type="text"
                      value="vab"
                      warn={false}
                      warnText=""
                    >
                      <div
                        className="bx--form-item bx--text-input-wrapper"
                      >
                        <div
                          className="bx--text-input__field-outer-wrapper"
                        >
                          <div
                            className="bx--text-input__field-wrapper"
                            data-invalid={null}
                          >
                            <input
                              className="bx--text-input bx--text-input--lg"
                              disabled={false}
                              id="test2"
                              name="lastName"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              placeholder="Last Name"
                              required={true}
                              title="Last Name"
                              type="text"
                              value="vab"
                            />
                          </div>
                        </div>
                      </div>
                    </ForwardRef(TextInput)>
                  </ForwardRef(Field)>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <Col
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col"
                >
                  <ForwardRef(Field)
                    name="gender"
                  >
                    <div
                      className="form__field"
                    >
                      <h4
                        htmlFor="male"
                      >
                        Gender
                      </h4>
                      <RadioButtonGroup
                        defaultSelected="male"
                        invalidText="This is an invalid error message."
                        labelPosition="right"
                        labelText="Last Name"
                        legend="Group Legend"
                        name="radio-button-group"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        orientation="horizontal"
                        value="male"
                        valueSelected="male"
                      >
                        <div
                          className="bx--form-item"
                        >
                          <fieldset
                            className="bx--radio-button-group bx--radio-button-group--label-right"
                          >
                            <ForwardRef(RadioButton)
                              checked={true}
                              id="male"
                              key="male/.0"
                              labelText="male"
                              name="radio-button-group"
                              onChange={[Function]}
                              value="male"
                            >
                              <RadioButton
                                checked={true}
                                id="male"
                                innerRef={null}
                                labelPosition="right"
                                labelText="male"
                                name="radio-button-group"
                                onChange={[Function]}
                                value="male"
                              >
                                <div
                                  className="bx--radio-button-wrapper"
                                >
                                  <input
                                    checked={true}
                                    className="bx--radio-button"
                                    id="male"
                                    name="radio-button-group"
                                    onChange={[Function]}
                                    type="radio"
                                    value="male"
                                  />
                                  <label
                                    className="bx--radio-button__label"
                                    htmlFor="male"
                                  >
                                    <span
                                      className="bx--radio-button__appearance"
                                    />
                                    <span
                                      className=""
                                    >
                                      male
                                    </span>
                                  </label>
                                </div>
                              </RadioButton>
                            </ForwardRef(RadioButton)>
                            <ForwardRef(RadioButton)
                              checked={false}
                              id="female"
                              key="female/.1"
                              labelText="female"
                              name="radio-button-group"
                              onChange={[Function]}
                              value="female"
                            >
                              <RadioButton
                                checked={false}
                                id="female"
                                innerRef={null}
                                labelPosition="right"
                                labelText="female"
                                name="radio-button-group"
                                onChange={[Function]}
                                value="female"
                              >
                                <div
                                  className="bx--radio-button-wrapper"
                                >
                                  <input
                                    checked={false}
                                    className="bx--radio-button"
                                    id="female"
                                    name="radio-button-group"
                                    onChange={[Function]}
                                    type="radio"
                                    value="female"
                                  />
                                  <label
                                    className="bx--radio-button__label"
                                    htmlFor="female"
                                  >
                                    <span
                                      className="bx--radio-button__appearance"
                                    />
                                    <span
                                      className=""
                                    >
                                      female
                                    </span>
                                  </label>
                                </div>
                              </RadioButton>
                            </ForwardRef(RadioButton)>
                            <ForwardRef(RadioButton)
                              checked={false}
                              id="another"
                              key="another/.2"
                              labelText="another"
                              name="radio-button-group"
                              onChange={[Function]}
                              required={true}
                              value="another"
                            >
                              <RadioButton
                                checked={false}
                                id="another"
                                innerRef={null}
                                labelPosition="right"
                                labelText="another"
                                name="radio-button-group"
                                onChange={[Function]}
                                required={true}
                                value="another"
                              >
                                <div
                                  className="bx--radio-button-wrapper"
                                >
                                  <input
                                    checked={false}
                                    className="bx--radio-button"
                                    id="another"
                                    name="radio-button-group"
                                    onChange={[Function]}
                                    required={true}
                                    type="radio"
                                    value="another"
                                  />
                                  <label
                                    className="bx--radio-button__label"
                                    htmlFor="another"
                                  >
                                    <span
                                      className="bx--radio-button__appearance"
                                    />
                                    <span
                                      className=""
                                    >
                                      another
                                    </span>
                                  </label>
                                </div>
                              </RadioButton>
                            </ForwardRef(RadioButton)>
                          </fieldset>
                        </div>
                      </RadioButtonGroup>
                    </div>
                  </ForwardRef(Field)>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            className="justify-content-center"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="justify-content-center row"
            >
              <Col
                className="d-flex form__field"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="d-flex form__field col"
                >
                  <Button
                    className="form__submit-btn false"
                    dangerDescription="danger"
                    disabled={false}
                    isExpressive={false}
                    kind="tertiary"
                    size="default"
                    tabIndex={0}
                    tooltipAlignment="center"
                    tooltipPosition="top"
                    type="submit"
                  >
                    <button
                      aria-describedby={null}
                      aria-pressed={null}
                      className="form__submit-btn false bx--btn bx--btn--tertiary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      tabIndex={0}
                      type="submit"
                    >
                      Submit
                    </button>
                  </Button>
                </div>
              </Col>
              <Col
                className="d-flex form__field"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="d-flex form__field col"
                >
                  <Button
                    className="form__reset-btn button--disabled bx--btn--danger--tertiary"
                    dangerDescription="danger"
                    disabled={true}
                    isExpressive={false}
                    kind="danger--tertiary"
                    onClick={[Function]}
                    size="default"
                    tabIndex={0}
                    tooltipAlignment="center"
                    tooltipPosition="top"
                    type="button"
                  >
                    <button
                      aria-describedby="danger-description-4"
                      aria-pressed={null}
                      className="form__reset-btn button--disabled bx--btn--danger--tertiary bx--btn bx--btn--danger--tertiary bx--btn--disabled"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="bx--visually-hidden"
                        id="danger-description-4"
                      >
                        danger
                      </span>
                      Reset
                    </button>
                  </Button>
                </div>
              </Col>
            </div>
          </Row>
        </form>
      </ReactFinalForm>
    </div>
  </Container>
</FormRequest>
`;
